core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -107,6 +107,17 @@ public Key(Text row) {
   }
 
   /**
+   * Creates a key with the specified row, empty column family, empty column qualifier, empty column visibility, timestamp {@link Long#MAX_VALUE}, and delete
+   * marker false.
```

Comment:
>  I wonder if this is the right behaviour or if the user would expect the byte array to be used as it is.

Copying the underlying array is definitely the safe/consistent thing to do. I think we should avoid making another set of constructors just to support users providing `copy=false`. That's pretty much what I was thinking about with ACCUMULO-4376.